### ISSUE TITLE
6.2 Reset API 401

### ISSUE THREAD

**Issue by xyzulu** [litespeedtech/lscache_wp#652]
I have resolved this by rolling back to v6.1, but v6.2 (clean install) gives me a 401 error hitting this page: /wp-json/litespeed/v1/token

I'm on OLS, but this appears all plugin related.

**Comment by xyzulu** [comment]
I understand your words, but how to resolve the issue on 6.2

**Comment by hi-hai** [comment]
It is caused by IPv6 issue. We have updated the QC side to v4 to fix this issue. Please try it again. Thanks

**Comment by xyzulu** [comment]
I've tried this again on a new site.. using 6.2.1 and I see the same behaviour as before.. (can't reach callback url.. check firewall etc). Rolled back to 6.1, requested domain key.. all good.

Am I missing something here?

PS This was a completely different site I tried this time.

**Comment by tjwebmaster** [comment]
I got the same problem. Disabled all plugins and checked if  REST API is enabled. Rolled back to 6.1 and now it is working again.

**Comment by xyzulu** [comment]
Yes, this behaviour still persists on 6.2 (fails). There is a discussion here on the issue too: https://wordpress.org/support/topic/cant-generate-quic-cloud-key-with-litespeed-6-2/

(up to 3 pages now)

**Comment by qtwrk** [comment]
if your site is behind proxy ,like a CDN service ,  and server didn't configure properly to forward the IP , like  PHP only sees proxy server IP , instead of true client IP , it could also lead to IP verification failed

**Comment by xyzulu** [comment]
@qtwrk my sites are.. as are most people's. What options do we have to work around this change that was made in 6.2 (things worked fine before).

I run sites behind Cloudflare in docker containers. Visitor IP can be correctly retrieved from the X-Forwarded-For header. Shouldn't Litespeed WP cache be using this? Something has changed in 6.2.. but I'm not a developer.. only a systems admin.

**Comment by qtwrk** [comment]
please go to LiteSpeed WebAdmin Console > Configuration > General Settings and set `Use Client IP in Header` to `Trusted IP Only`

**Comment by tjwebmaster** [comment]
I think a lot of people don't have access to the WebAdmin console (including me) so that cant be the solution right?

**Comment by jstubbs34** [comment]
Ok, so we are having the same issue. Before 6.2 everything was working fine but now none of our customers can get domain keys and even rolling back the plugin to 6.1 doesn't get the domain key. I'm not sure what you guys did but it worked now it doesn't and yes our LiteSpeed has Use Client IP in Header set to Trusted IP Only. This is pretty frustrating because nothing we have tried is working and our customers are getting pissed which is totally understandable considering things were working fine until whatever you changed broke everything. It's almost worth dropping LiteSpeed and just rolling out nginx with Bunny CDN because this is just crazy to break so many servers and to piss off so many customers over some change concerning IP's. Like the saying goes if it isn't broken don't try to fix it. LiteSpeed ran fine now it's just an expensive software that does nothing I can use a free software like nginx and get the same result and pocket the cash we are spending on LSWS.

**Comment by qtwrk** [comment]
if you still have issue on v6.1 , then it must be something else gets the way 

please create a [ticket](https://store.litespeedtech.com/store/submitticket.php) , we will check further

**Comment by jstubbs34** [comment]
> if you still have issue on v6.1 , then it must be something else gets the way
> 
> please create a [ticket](https://store.litespeedtech.com/store/submitticket.php) , we will check further

Not sure how that could be considering the accounts on that server we all connected to QUIC.cloud until version 6.2 now none connect rolling back. Nothing about the server was changed so if nothing about the server has changed and everything worked fine until 6.2 then not sure what else could get in the way. Also the plugin seems to me malfunctioning because even the debug logs don't appear when enabled.

**Comment by leandrocastro** [comment]
Solution
https://wordpress.org/support/topic/domain-key-issue-on-v6-2/

**Comment by BreezeMaxWeb-NS** [comment]
I started seeing this issue on OpenLiteSpeed a few weeks ago, and now I'm seeing it on LiteSpeed with the latest plugin version (7.7).

I have set:
Use Client IP in Header | Trusted IP Only

And to handle the Cloudflare CDN IPs I've set trusted:
Allowed List:
```
ALL, 127.0.0.1T, 173.245.48.0/20T, 103.21.244.0/22T, 103.22.200.0/22T, 103.31.4.0/22T, 141.101.64.0/18T, 108.162.192.0/18T, 190.93.240.0/20T, 188.114.96.0/20T, 197.234.240.0/22T, 198.41.128.0/17T, 162.158.0.0/15T, 104.16.0.0/13T, 104.24.0.0/14T, 172.64.0.0/13T, 131.0.72.0/22T, 2400:cb00::/32T, 2606:4700::/32T, 2803:f800::/32T, 2405:b500::/32T, 2405:8100::/32T, 2a06:98c0::/29T, 2c0f:f248::/32T
```

This has worked for years with the same configuration across many different servers, dedicated and vps, but has suddenly become a problem and I can't get a new domain key connection, `QUIC.cloud's access to your WP REST API seems to be blocked.`.

Note, WP REST API has been verified working as expected.
